src/main/java/com/cloudbees/jenkins/plugins/bitbucket/filesystem/BitbucketSCMFileSystem.java

```diff
@@ -176,8 +176,9 @@ public SCMFileSystem build(@NonNull SCMSource source, @NonNull SCMHead head, @Ch
                         return null;
                     }
                 } else if (pr.getCheckoutStrategy() == ChangeRequestCheckoutStrategy.HEAD) {
-                    ref = "pull-requests/" + pr.getId() + "/from";
+                    ref = "heads/" + pr.getBranchName();
                 } else if (pr.getCheckoutStrategy() == ChangeRequestCheckoutStrategy.MERGE) {
+                    // No longer supported since bitbucket server v7, falls back to heavycheckout
                     ref = "pull-requests/" + pr.getId() + "/merge";
                 }
```

Comment:
@Jetersen What do you think? This supportes older bitbucket and falls back to heavy checkout on bitbucket v7 and later

Reply:
Yup seems fine with me, properly simpler fix 😅

Reply:
Older versions of BitBucket, while they had the pull-request refs, are not accurate. PRs get confused in our environment, and we're running 6.7.2.

Reply:
So the pr refs are not updated at 6.7.2 too?

Reply:
Not without performing a 'null' operation on the UI, such as editing and saving the PR without making any changes.

Reply:
or using the bitbucket webhook plugin that was actually doing this work for you.

https://marketplace.atlassian.com/apps/1215474/post-webhooks-for-bitbucket?hosting=server&tab=overview

Reply:
I'd recommend getting rid of the use of "refs/pull-requests" globally, as it was never a supported API by BitBucket, and only worked by accident or as a side effect, apparently.

Reply:
it has been working great for others so I have to disagree.

Reply:
Alright, but it did stop working before v7.  Not sure when.

Reply:
Would exposing a configuration option be a compromise?

Reply:
this condition only allows jenkins to do a lightweight checkout on master to fetch the jenkins file. When doing a full checkout on worker it would always fetch the correct commits with this pr.

So there is only a small window when jenkins fetches an out of date jenkinsfile. So this can only happen, if you push changes to an existing pr, which updates the jenkinsfile

Reply:
Oh, my mistake then.

Reply:
I think this comment (https://jira.atlassian.com/browse/BSERV-12284?focusedCommentId=2389584&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2389584) is a quite interesting read on the topic.
Taking this into account I would also get rid of that merge ref as it does not seem to come back and it can be outdated on earlier Bitbucket versions if people don't use special Bitbucket plugins that trigger the update of the refs, e.g. when using only the native web-hooks.
I don't know this source code well enough but we might want to add that REST API call (rest/api/1.0/projects/KEY/repos/slug/pull-requests/N/changes?limit=1) to make sure the from ref is properly created/updated if we still want to use it. This would allow users to stick with the native web-hooks solution without any additional plugins.
Thoughts?